src/commands/moderation/kick.ts

```diff
@@ -0,0 +1,76 @@
+import {
+  highestRole,
+  higherRolePosition,
+  Member,
+  botID,
+  sendDirectMessage,
+} from "../../../deps.ts";
+import { botCache } from "../../../mod.ts";
+import { createCommandAliases } from "../../utils/helpers.ts";
+import { PermissionLevels } from "../../types/commands.ts";
+import { kick } from "https://x.nest.land/Discordeno@8.4.4/src/handlers/member.ts";
+
+botCache.commands.set(`kick`, {
+  name: `kick`,
+  permissionLevels: [PermissionLevels.MODERATOR, PermissionLevels.ADMIN],
+  botServerPermissions: ["KICK_MEMBERS"],
+  arguments: [
+    { name: "member", type: "member", required: false },
+    { name: "userID", type: "snowflake", required: false },
+    { name: "reason", type: "...string" },
+  ],
+  guildOnly: true,
+  execute: async function (message, args: KickArgs, guild) {
+    if (!guild) return;
+
+    if (args.member) {
+      const botsHighestRole = highestRole(message.guildID, botID);
+      const membersHighestRole = highestRole(
+        message.guildID,
+        args.member.user.id,
+      );
+      const modsHighestRole = highestRole(message.guildID, message.author.id);
+
+      if (
+        !botsHighestRole || !membersHighestRole ||
+        !higherRolePosition(
+          message.guildID,
+          botsHighestRole.id,
+          membersHighestRole.id,
+        )
+      ) {
+        return botCache.helpers.reactError(message);
+      }
+
+      if (
+        !modsHighestRole || !membersHighestRole ||
+        !higherRolePosition(
+          message.guildID,
+          modsHighestRole.id,
+          membersHighestRole.id,
+        )
+      ) {
+        return botCache.helpers.reactError(message);
+      }
+    }
+
+    const userID = args.member?.user.id || args.userID!;
```

Comment:
args.member will always exist and args.userID will never exist so we can just delete this line.